test/Microsoft.ML.Tests/PermutationFeatureImportanceTests.cs

```diff
@@ -148,47 +271,42 @@ public void TestSparseSGD()
             var pipeline = ML.Transforms.Concatenate("Features", "X1", "X2VBuffer", "X3Important")
                 .Append(ML.Transforms.Normalize("Features"));
             var data = pipeline.Fit(srcDV).Transform(srcDV);
-            var model = ML.Regression.Trainers.OnlineGradientDescent().Fit(data);
-            var results = ML.Regression.PermutationFeatureImportance(model, data);
-
-            // Pfi Indices:
-            // X1: 0
-            // X2VBuffer-Slot-0: 1
-            // X2VBuffer-Slot-1: 2
-            // X2VBuffer-Slot-2: 3
-            // X2VBuffer-Slot-3: 4
-            // X3Important: 5
 
-            // Permuted X2VBuffer-Slot-1 lot (f2) should have min impact on SGD metrics, X3Important -- max impact.
-            // For the following metrics lower is better, so maximum delta means more important feature, and vice versa
-            Assert.True(MinDeltaIndex(results, m => m.L1) == 2);
-            Assert.True(MaxDeltaIndex(results, m => m.L1) == 5);
-
-            Assert.True(MinDeltaIndex(results, m => m.L2) == 2);
-            Assert.True(MaxDeltaIndex(results, m => m.L2) == 5);
-
-            Assert.True(MinDeltaIndex(results, m => m.Rms) == 2);
-            Assert.True(MaxDeltaIndex(results, m => m.Rms) == 5);
-
-            // For the following metrics higher is better, so minimum delta means more important feature, and vice versa
-            Assert.True(MaxDeltaIndex(results, m => m.RSquared) == 2);
-            Assert.True(MinDeltaIndex(results, m => m.RSquared) == 5);
+            return data;
         }
 
-        private int MinDeltaIndex(
-            ImmutableArray<RegressionEvaluator.Result> metricsDelta,
-            Func<RegressionEvaluator.Result, double> metricSelector)
+        private int MinDeltaIndex<T>(
+            ImmutableArray<T> metricsDelta,
+            Func<T, double> metricSelector)
         {
             var min = metricsDelta.OrderBy(m => metricSelector(m)).First();
             return metricsDelta.IndexOf(min);
         }
 
-        private int MaxDeltaIndex(
-            ImmutableArray<RegressionEvaluator.Result> metricsDelta,
-            Func<RegressionEvaluator.Result, double> metricSelector)
+        private int MaxDeltaIndex<T>(
+            ImmutableArray<T> metricsDelta,
+            Func<T, double> metricSelector)
         {
             var max = metricsDelta.OrderByDescending(m => metricSelector(m)).First();
             return metricsDelta.IndexOf(max);
         }
+
+        private void GetBinaryClassificationScores(float[] rawScores)
+        {
+            float averageScore = 0.0f;
+            for (int i = 0; i < rawScores.Length; i++)
+                averageScore += rawScores[i];
+            averageScore /= rawScores.Length;
```

Comment:
I go back and forth on Linq. It's about 5-10 times slower, so I don't like to put it in this codebase, because someone may copy and paste it out.

As for overflow, it doesn't do anything special, other than run the compute in a `checked` block. Since we are only using this for summing a small number of things, I think we're okay using an `unchecked` summation here.

Unless you feel strongly, I'll keep the hand-calculated average.